src/Microsoft.Azure.Batch.SoftwareEntitlement.Common/Specifiable.cs

```diff
@@ -41,5 +43,39 @@ public Specifiable(T value)
         /// <paramref name="other"/> value.
         /// </returns>
         public T OrDefault(T other) => IsSpecified ? _value : other;
+
+        /// <summary>
+        /// Test to see if another <see cref="Specifiable{T}"/> is equal to this instance
+        /// </summary>
+        /// <param name="other">Other specifiable for comparison.</param>
+        /// <returns>True if both contain the same value, false otherwise.</returns>
+        public bool Equals(Specifiable<T> other)
+            => ReferenceEquals(_value, other._value)
+               || (_value?.Equals(other._value) ?? false);
```

Comment:
I'd just use `=> Equals(_value, other._value);` here. This will correctly handle `null`s.

Reply:
https://referencesource.microsoft.com/#mscorlib/system/object.cs,68